stl/inc/xutility

```diff
@@ -6070,6 +6072,114 @@ private:
     _Udiff _Bmask; // 2^_Bits - 1
 };
 
+template <class _Diff, class _Urng>
+class _Rng_from_urng_v2 { // wrap a URNG as an RNG
+public:
+    using _Ty0 = make_unsigned_t<_Diff>;
+    using _Ty1 = _Invoke_result_t<_Urng&>;
+
+    using _Udiff                              = conditional_t<sizeof(_Ty1) < sizeof(_Ty0), _Ty0, _Ty1>;
+    static constexpr unsigned int _Udiff_bits = sizeof(_Udiff) * CHAR_BIT;
+    using _Uprod = conditional_t<_Udiff_bits <= 16, uint32_t, conditional_t<_Udiff_bits <= 32, uint64_t, _Unsigned128>>;
+
+    explicit _Rng_from_urng_v2(_Urng& _Func) : _Ref(_Func) {}
+
+    _Diff operator()(_Diff _Index) { // adapt _Urng closed range to [0, _Index)

```

Comment:
I'm not explicitly asking for this (I don't know if my fellow maintainers would agree), but it might be nice to switch out these names for names more closely matching the paper? Something similar to what we do in the ryu files...

Reply:
Alternatively, comments on the lines which line up to the original implementation; something like

```cxx
// 2**L
_Udiff _Mask        = _Bmask;
...
// x <- random integer in [0, 2**L)
// m <- x * s
auto _Product = _Get_random_product(_Index, _Niter);
// l <- m mod 2**L
auto _Rem = static_cast<_Udiff>(_Product) & _Mask;

Reply:
I'd be more amenable to comments, because single-letter variable names like `_L` are disfavored.

Reply:
Agreed that comments would be nice.